Feature/auth

vlapey opened the pull request.

vlapey committed:
Merge pull request #8 from vlapey/feature/mixins

Feature/mixins

vlapey committed:
dropped extra "cars" file

vlapey committed:
1. registered all apps
2. renamed DealerCar to DealersCar in dealer model
3. added extra src prefixes

vlapey committed:
1. registered all apps
2. added src where needed
3. renamed Car class to Cars
4. added django rest swagger to requirements.txt
5. added urls for viewsets
6. changed GenericApiView to GenericViewSet

Chaddi13 commented on docker-compose.yml:
```diff
@@ -0,0 +1,30 @@
+    build: .
+    ports:
+      - "8000:8000"
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postgres
+      - POSTGRES_HOST=db
+      - POSTGRES_PORT=5432
```
.env

Chaddi13 commented on docker-compose.yml:
```diff
@@ -0,0 +1,30 @@
+  db:
+    container_name: db
+    image: postgres:12
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=postgres
```
.env

Chaddi13 commented on src/cars/models.py:
```diff
@@ -0,0 +1,21 @@
+
+class CarSpecification(ModelProperties):
+    name = models.CharField(max_length=30, unique=True)
+    engine = models.CharField(max_length=60)
+    horsepower = models.IntegerField(default=0)
+    torque = models.IntegerField(default=0)
```
Не могут быть отрицательными.

vlapey committed:
added cat_type and color fields to Cars model

vlapey committed:
overrided create method for car serializer to create specification if it doesn't exist

vlapey committed:
made horsepower and torque only positive

vlapey committed:
put docker-compose variables to .env file

vlapey committed:
changed .env.local so

vlapey committed:
1. added djangorestframework-simplejwt to requirements.txt
2. added jwt settings to settings.py
3. added auth to urls

vlapey committed:
1. added Profiles in enums for permissions
2. added custom permission
3. refactored car model (put brand and model into car specification, added mileage to car)
4. added abstract class to loyalties to follow DRY
5. added swagger and django-debug-tools
6. refactored serializers to exclude is_active field
7. deleted overrided create method in car serializer
8. added permissions for car and customer

vlapey committed:
added django-filter to requirements.txt

vlapey committed:
Merge branch 'develop' into feature/auth
